pkg/runtime/runtime.go

```diff
@@ -46,7 +47,16 @@ func New(path string) (*Runtime, error) {
 		return nil, errs.Wrap(err, "Could not create runtime directory")
 	}
 
-	depot, err := newDepot()
+	// Windows does not support hard-linking across drives, so determine if the runtime path is on a
+	// separate drive than the default depot path. If so, use a drive-specific depot path when
+	// initializing the depot.
+	runtimeVolume := filepath.VolumeName(path)
+	storageVolume := filepath.VolumeName(storage.CachePath())
+	if runtimeVolume == storageVolume {
+		runtimeVolume = ""
+	}
```

Comment:
I think this logic can live within depot, no need for runtime to deal with these concerns.

Reply:
Also does this need to be windows specific logic? What does `filepath.VolumeName()` do on non-windows systems? Probably safer to avoid using this logic outside of windows, even if the APIs are satisfied.

Reply:
The documentation says that `filepath.VolumeName()` returns the empty string for non-Windows OSes. However, I can wall it off behind a `runtime.GOOS` check.